timeseries/src/autogluon/timeseries/models/autogluon_tabular/transforms.py

```diff
@@ -0,0 +1,60 @@
+from typing import Literal, Optional, Union
+
+import numpy as np
+import pandas as pd
+from mlforecast.target_transforms import (
+    BaseTargetTransform,
+    GroupedArray,
+    _BaseGroupedArrayTargetTransform,
+)
+
+from autogluon.timeseries.dataset.ts_dataframe import (
+    ITEMID,
+    TIMESTAMP,
+    TimeSeriesDataFrame,
+)
+from autogluon.timeseries.transforms.scaler import LocalTargetScaler, get_target_scaler
+
+from .utils import MLF_ITEMID, MLF_TIMESTAMP
+
+
+class MLForecastScaler(BaseTargetTransform):
+    def __init__(self, scaler_type: Literal["standard", "robust", "min_max", "mean_abs"]):
+        # For backward compatibility
+        self.scaler_type = scaler_type
+        self.ag_scaler: Optional[LocalTargetScaler] = None
+
+    def _df_to_tsdf(self, df: pd.DataFrame) -> TimeSeriesDataFrame:
+        return df.rename(columns={self.id_col: ITEMID, self.time_col: TIMESTAMP}).set_index([ITEMID, TIMESTAMP])
+
+    def _tsdf_to_df(self, ts_df: TimeSeriesDataFrame) -> pd.DataFrame:
+        return pd.DataFrame(ts_df).reset_index().rename(columns={ITEMID: self.id_col, TIMESTAMP: self.time_col})
+
+    def fit_transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        self.ag_scaler = get_target_scaler(name=self.scaler_type, target=self.target_col)
+        transformed = self.ag_scaler.fit_transform(self._df_to_tsdf(df)).reset_index()
+        return self._tsdf_to_df(transformed)
+
+    def inverse_transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        transformed = self.ag_scaler.inverse_transform(self._df_to_tsdf(df))
+        return self._tsdf_to_df(transformed)
+
+
+def apply_inverse_transform(
```

Comment:
I was a bit surprised at first about this function not being inside of the class above, but I suppose that this comes from the fact that we need to handle both `BaseTargetTransform` and `_BaseGroupedArrayTargetTransform`?

Reply:
Exactly, we need a function that works both with `BaseTargetTransform` and `_BaseGroupedArrayTargetTransform` classes that have different expectations for the input format: the former works on `pd.DataFrame` while the latter only works on `GroupedArray`.